src/main/java/com/charlesgadeken/entwined/EngineController.java

```diff
@@ -0,0 +1,146 @@
+package com.charlesgadeken.entwined;
+
+import com.charlesgadeken.entwined.effects.TSEffectController;
+import com.charlesgadeken.entwined.effects.original.ScrambleEffect;
+import com.charlesgadeken.entwined.effects.original.SpeedEffect;
+import com.charlesgadeken.entwined.effects.original.SpinEffect;
+import heronarts.lx.LX;
+import heronarts.lx.effect.BlurEffect;
+import heronarts.lx.mixer.LXAbstractChannel;
+import heronarts.lx.mixer.LXChannel;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class EngineController {
+    LX lx;
+
+    public int baseChannelIndex;
+    public int numChannels;
+
+    public int startEffectIndex;
+    public int endEffectIndex;
+
+    public boolean isAutoplaying;
+    EntwinedAutomationRecorder automation;
+    boolean[] previousChannelIsOn;
+
+    public List<TSEffectController> effectControllers = new ArrayList<>();
+    public int activeEffectControllerIndex = -1;
+
+    public SpeedEffect speedEffect;
+    public SpinEffect spinEffect;
+    public BlurEffect blurEffect;
+    public ScrambleEffect scrambleEffect;
+
+    public EngineController(LX lx) {
+        this.lx = lx;
+    }
+
+    public List<LXChannel> getChannels() {
+        // NOTE(meawoppl) dangercast here. @Slee?
+        return lx.engine.mixer.getChannels()
+                .subList(baseChannelIndex, baseChannelIndex + numChannels).stream()
+                .map(c -> (LXChannel) c)
+                .collect(Collectors.toList());
+    }
+
+    public void setChannelPattern(int channelIndex, int patternIndex) {
+        // NOTE(meawoppl) - Needs review from @Slee?
+        // Previously this was:
+        //        if (patternIndex == -1) {
+        //            patternIndex = 0;
+        //        } else {
+        //            patternIndex++;
+        //        }
+        //        lx.engine.getChannel(channelIndex).goIndex(patternIndex);
+        // Trying....
+        lx.engine.mixer.setFocusedChannel(lx.engine.mixer.getChannel(channelIndex));
+        // ??
+    }
+
+    public void setChannelVisibility(int channelIndex, double visibility) {
```

Comment:
Oof, makes sense for backwards compatibility but it is weird to be overloading methods with different terminology. Not a big deal but "visibility" feels like an odd choice for fader level.

Reply:
Yeah, this def. isn't my codebase, there is quite a bit of weird running around.
I think I am the 3rd generation to pick this up and try to make it do something useful now ;p